### DISCUSSION TITLE
Add DynMap (IF POSSIBLE)

### DISCUSSION THREAD

**Discussion by expxx** [Minehut/Meta#535] [General]
Add the dynmap plugin, I just want to know if it would be humanly possible for it to work on minehut.

**Comment by quick007** [comment]
I'm perssonnaly upvoting this, I know we discussed something like servername.mhmap.gg, but idk. 

just gonna go ahead and leave this here:

### Dynmap for servers

I feel that for many survival servers, as well as creative and mcmmorpgs lack someone that I would consider to be a preatty key part of any server. Dynmaps allow people to see the map, there location, and other helpful information that is very useful on many types of servers. Unfortunately, minehut doesn't have this plugin. I can partially see why, but I feel that there would be a really good way for servers to easily have a map. Before you downvote this, please make sure to read the whole post.

**Execution**

Hands down, the best way for links to work would be though a minehut link. People could possibly get a custom domain (maybe mh uses g domains and take a cut of the price paid? idk.) My idea for a domain would be <servername>.map.minehut.gg. This allows for a free link for minehut, whilst still making it easy for people to access it. 

I think that another tab on the panel that would allow people to select worlds (and possibly which part of them, i.e. a 100x100 aquare around spawn). Secondly, they would be able to select whether players, minehut friends, themselves, or no one can be seen on the map. (these are all the options I can think of atm, but would be nice to hear others ideas in #discussion)

**Paywalls**

I'm not a big fan of paywalls, but, judging from where mh is going, a project like this is never going to be put into action/upkept without monetary incentive. Heres my idea on how to handle it:

Server owners that don't want to pay for it, can take "snapshots" of there world which would allow people to get a birds eye view of there map. Could be online, could not be. If its online, maybe it could transmit player loc data as well? Would take a cooldown to be able to get snapshots.

Paying people (I was thinking mh35 or 75 and above would get it free) would get the full dynmap experiance, with the link nd everything. Possibility for people to pay a monthly fee for it if they don't have a high plan.

**Comment by expxx** [comment]
> Hands down, the best way for links to work would be though a minehut link. People could possibly get a custom domain (maybe mh uses g domains and take a cut of the price paid? idk.) My idea for a domain would be .map.minehut.gg. This allows for a free link for minehut, whilst still making it easy for people to access it.

Maybe, because some users have a custom domain, there’s an option to setup a custom domain for it. If you don’t have a domain, or just don’t want to use theirs, you use the `.map.Minehut.gg` link..?

**Comment by ghost** [comment]
I don't see why this wouldn't be a good idea. I like it.

**Comment by Sniped** [comment]
Plugin suggestions won't be looked at here, please suggest this plugin on the [forums](https://forums.minehut.com/forum/20-plugins/) and have discussions pertaining to DynMap there. 

Looking through the threads on the forums, however, it seems like a great number of people have already suggested DynMap and it's already been shot down by pop. (refer to the screenshot below)

![](https://i.imgur.com/egNLMQ6.png)

**Comment by wiidotmom** [comment]
Adding onto this, even if it's not on the Minehut forums where pop checks: It's not totally impossible for Minehut to build their own Dynmap solution. Dynmap has an option to write its data to a specified path on the same machine instead of the local webserver. Theoretically, it'd be possible to use a reverse proxy such as nginx to spit out the web files for anyone who makes a request to, for example, `mycoolserver123.minehut.gg/dynmap`. I have no idea how Minehut's backend is actually structured, but theoretically, it's possible to do it if Minehut doesn't want to port-forward the internal webserver.

**Comment by expxx** [comment]
> Plugin suggestions won't be looked at here, please suggest this plugin on the [forums](https://forums.minehut.com/forum/20-plugins/) and have discussions pertaining to DynMap there.
> 
> Looking through the threads on the forums, however, it seems like a great number of people have already suggested DynMap and it's already been shot down by pop. (refer to the screenshot below)
> 
> ![](https://camo.githubusercontent.com/1825434ad978dd7f67a0372867ce3b685babd0f56c5f55e199f1da690a17a13b/68747470733a2f2f692e696d6775722e636f6d2f65674e4c4d51362e706e67)

I never wanted it to be a plugin suggestion. Just if it’s possible, and how it might be done.

**Comment by Sniped** [comment]
> I never wanted it to be a plugin suggestion. Just if it’s possible, and how it might be done.

Then this should've been a discussion rather than a feature suggestion.

**Reply by Sniped** [reply]
Converted this to a discussion

**Reply by expxx** [reply]
Better :)

**Comment by expxx** [comment]
> > I never wanted it to be a plugin suggestion. Just if it’s possible, and how it might be done.
> 
> Then this should've been a discussion rather than a feature suggestion.

Idk how to create discussions 🤷

**Comment by expxx** [comment]
Otherwise I would have
